### PR TITLE
Add AL1 keyboard to QMK

### PR DESCRIPTION
**Pull request by olivia**
- [x] Move over folder from AL1 Repo
- [ ] Add instructions to flash since default keymap is missing reset key
- [ ] Add better keymap with reset key

### CODE REVIEW

File: keyboards/al1/al1.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#define AL1_H
+
+#include "quantum.h"
+
+#define CONFIG_LED_IO \
+  DDRB |= (1<<7) | (1<<6); \
+  DDRD |= (1<<0) | (1<<1);
+
+
+#define KEYMAP( \
```
Comment (mechmerlin) [inline keyboards/al1/al1.h]: Can you rename `KEYMAP` to `LAYOUT_all` in this .h file and in the keymap.c files that reference it?

File: keyboards/al1/al1.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+{ \
+    { K000, K001, K002, K003, K004, K005, K006, K007, K008, K009, K010, K011, K012, K013, K014, K015}, \
+    { K100, K101, K102, K103, K104, K105, K106, K107, K108, K109, K110, K111, K112, K113, K114, K115}, \
+    { K200, K201, K202, K203, K204, K205, K206, K207, K208, K209, K210, K211, K212, K213, K214, K215}, \
+    { K300, K301, K302, K303, K304, K305, K306, K307, K308, K309, K310, K311, K312, K313, K314, K315}, \
+    { K400, K401, K402, K403, K404, K405, K406, K407, K408, K409, K410, K411, K412, K413, K414, K415}, \
+    { K500, K501, K502, K503, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO} \
+}
+
+#define KEYMAP_SPLITBS( \
```
Comment (mechmerlin) [inline keyboards/al1/al1.h]: Can you rename this to `LAYOUT_split_bs` and in the keymap.c files that reference it?

File: keyboards/al1/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,56 @@
+#include "al1.h"
```
Comment (mechmerlin) [inline keyboards/al1/keymaps/default/keymap.c]: Please replace `#include "all.h` with `#include QMK_KEYBOARD_H`

File: keyboards/al1/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+       [0] = KEYMAP(\
```
Comment (mechmerlin) [inline keyboards/al1/keymaps/default/keymap.c]: Change `KEYMAP` to `LAYOUT_all` and its other references in the file

File: keyboards/al1/keymaps/splitbs/keymap.c [hunk 5]
```diff
@@ -0,0 +1,56 @@
+#include "al1.h"
```
Comment (mechmerlin) [inline keyboards/al1/keymaps/splitbs/keymap.c]: Change to `#include QMK_KEYBOARD_H`

File: keyboards/al1/keymaps/splitbs/keymap.c [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+	[0] = KEYMAP_SPLITBS(\
```
Comment (mechmerlin) [inline keyboards/al1/keymaps/splitbs/keymap.c]: Change to `LAYOUT_split_bs` and its other references in this file too!

**Comment by drashna** [comment]
Is the custom matrix necessary here?  If not, then it may be best to use the default one.

As for the bootloader, if you have a "RESET" code in the keymap, you should be okay.  
Otherwise, make sure that BootMagic is enabled, as holding SPACE and B while plugging in the keyboard will jump to the bootloader, instead of QMK. 
https://docs.qmk.fm/#/feature_bootmagic

File: keyboards/al1/rules.mk [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+F_CPU = 16000000
+
+ARCH = AVR8
+
+F_USB = $(F_CPU)
+
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (drashna) [inline keyboards/al1/rules.mk]: Do we know what controller is being used here? 
Eg, is this a Pro Micro, Teensy 2.0, other, or SMD?

**Comment by drashna** [comment]
#3330 supersedes this

**Pull request closed without merging**